Gems/ONNX/Code/Tests/MnistTests.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Mnist.h>
+#include <ONNXSystemComponent.h>
+
+#include <AzCore/Component/ComponentApplicationBus.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/Settings/SettingsRegistryImpl.h>
+#include <AzCore/Settings/SettingsRegistryMergeUtils.h>
+#include <AzCore/Slice/SliceAssetHandler.h>
+#include <AzFramework/IO/LocalFileIO.h>
+#include <AzTest/AzTest.h>
+#include <AzTest/GemTestEnvironment.h>
+#include <AzToolsFramework/Commands/PreemptiveUndoCache.h>
+#include <AzToolsFramework/Entity/EditorEntityContextComponent.h>
+#include <AzToolsFramework/ToolsComponents/TransformComponent.h>
+#include <QApplication>
+#include <UnitTest/ToolsTestApplication.h>
+
+namespace Mnist
+{
+    class MnistTestEnvironment : public AZ::Test::GemTestEnvironment
+    {
+        void AddGemsAndComponents() override
+        {
+            AddComponentDescriptors({ ::ONNX::ONNXSystemComponent::CreateDescriptor() });
+
+            AddRequiredComponents({ ::ONNX::ONNXSystemComponent::TYPEINFO_Uuid() });
+        }
+
+        void PreCreateApplication() override
+        {
+            m_fileIO = AZStd::make_unique<AZ::IO::LocalFileIO>();
+            AZ::IO::FileIOBase::SetInstance(m_fileIO.get());
+        }
+
+        void PostDestroyApplication() override
+        {
+            AZ::IO::FileIOBase::SetInstance(nullptr);
+            m_fileIO.reset();
+        }
+
+    private:
+        AZStd::unique_ptr<AZ::IO::LocalFileIO> m_fileIO;
+    };
+
+    class MnistFixture : public ::testing::Test
+    {
+    protected:
+        void SetUp() override
+        {
+            AZ::SettingsRegistryImpl localRegistry;
+            localRegistry.Set(AZ::SettingsRegistryMergeUtils::FilePathKey_EngineRootFolder, AZ::Test::GetEngineRootPath());
+
+            // Look up the path to the ONNX Gem Folder (don't assume it is in the Engine root)
+            // via searching through the gem paths that are registered in the o3de manifest files
+            // Adding the ONNX gem as an active gem allows the alias of @gemroot:ONNX@ to be
+            // set in the fileIO
+            AZ::SettingsRegistryMergeUtils::MergeSettingsToRegistry_ManifestGemsPaths(localRegistry);
+            AZ::Test::AddActiveGem("ONNX", localRegistry, AZ::IO::FileIOBase::GetInstance());
```

Comment:
This looks good.
This makes sure the ONNX gem folder can be located regardless of where it is located on the filesystem, even if it is external to the engine directory.